### DISCUSSION TITLE
ZSH .zshenv and pyenv

### DISCUSSION THREAD

**Discussion by ghost** [pyenv/pyenv#2069] [Q&A] [answered]
I  added `eval (pyenv init -)`  to .zshenv on macOS and am running into an issue. Can someone explain what might be causing this?

```
$nano .zshenv
```
I am getting this notice in Terminal once
```
$exec "$SHELL"
/Users/john/.zshenv:1: unknown file attribute: y
```

**Comment by native-api** [comment, marked as answer]
> I added `eval (pyenv init -)` to .zshenv

That's not what the README says to do.

Also check https://github.com/pyenv/pyenv/pull/2067 for soon-to-be-added (as of this writing) and hopefully clearer instructions.

**Reply by ghost** [reply]
I know but I am running into a conflict with brew and configuration files and trying to work around that.

Is there a pyenv approved method to avoid getting the brew doctor warnings regarding configuration files in the path?

**Reply by native-api** [reply]
Whatever you're doing, the issue that you've reported comes from an incorrect invocation.

**Reply by native-api** [reply]
> Is there a pyenv approved method to avoid getting the brew doctor warnings regarding configuration files in the path?

If you report _that_ issue, with reporoduction steps, we'll see what we can do.

**Reply by ghost** [reply]
I am just referencing the "solution" that another user, who experienced the same issue, found worked. It's from a few years back but it's the same issue - so this is undoubtedly a known issue and me adding it again won't help.

https://binx.io/blog/2019/04/12/installing-pyenv-on-macos/

**Reply by native-api** [reply]
> 
> 
> I am just referencing the "solution" that another user, who experienced the same issue, found worked. It's from a few years back but it's the same issue - so this is undoubtedly a known issue and me adding it again won't help.
> 
> https://binx.io/blog/2019/04/12/installing-pyenv-on-macos/

https://github.com/pyenv/pyenv/issues/1722

**Reply by ghost** [reply]
> Also check #2067 for soon-to-be-added (as of this writing) and hopefully clearer instructions.

I used those instructions and that results in my path looking like this:

```
/usr/local/opt/nano/bin
/usr/local/sbin
/Users/john/.pyenv/shims
/usr/local/bin
/usr/bin
/bin
/usr/sbin
/sbin
/opt/X11/bin
/Library/Apple/usr/bin
/Applications/Wireshark.app/Contents/MacOS
```

Isn't the shims directory supposed to be at the very top of my path?

```
$which python
/Users/john/.pyenv/shims/python

$python -V
Python 3.9.7

$brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/john/.pyenv/shims/python3.9-config
  /Users/john/.pyenv/shims/python-config
  /Users/john/.pyenv/shims/python3-config
```

There is no resolution to #1722 that I can tell. It just trails off with a bunch of different suggestions over a 5 year timespan and the issue is closed.

**Reply by native-api** [reply]
> Isn't the shims directory supposed to be at the very top of my path?

Normally, yes. This means that you have smth else in your shell configuration that runs later and adds those entries.

I guess you have some PATH manipulations in `~/.zshrc`. They should go into `~/.zprofile` instead.


> There is no resolution to #1722 that I can tell. It just trails off with a bunch of different suggestions over a 5 year timespan and the issue is closed.

[There is, in the issue that it's closed as a duplicate of](https://github.com/pyenv/pyenv/issues/106#issuecomment-779325818).
